### PR TITLE
chore: cleanup and upgrade

### PR DESCRIPTION
**Pull request by arxeiss**
# Checklist

- [x] `make test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/indykite/.github/blob/master/commit-message.md)
- [ ] is breaking change

## Description of change

Upgrading dependencies and cleaning the code. There are lots of changes related to
- Changed test to use Ginkgo and Gomega
- Apply linter rules
- Improve error handling to report some errors as Plugin errors

### CODE REVIEW

**Review by bakkelim — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -36,25 +36,25 @@ repos:
...
 
     - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt.git
-      rev: 0.0.10
+      rev: 0.2.2
       hooks:
           - id: yamlfmt
 
     - repo: https://github.com/jumanjihouse/pre-commit-hooks.git
-      rev: 2.1.6 # Do not update to 3.0.0 unless macOS comes with Ruby 2.7
+      rev: 3.0.0 # Do not update to 3.0.0 unless macOS comes with Ruby 2.7
```
Comment (marvinkite) [inline .pre-commit-config.yaml:44]: This shell be downgraded.

**Review by marvinkite — CHANGES_REQUESTED** [review]
Everything looks good, only one little thing for OSX users with older ruby. We can't update that pre-commit plugin.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (arxeiss) [reply .pre-commit-config.yaml:44]: Yep, automated upgrade. Will revert it back.

Comment (arxeiss) [reply .pre-commit-config.yaml]: Reverted back

**Review by marvinkite — APPROVED** [review]

**Pull request merged**